resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/FactoryFinder.java

```diff
@@ -113,20 +115,26 @@ static Object find(final String factoryId, final String fallbackClassName) throw
            }
 
            if (is != null) {
-               BufferedReader rd = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+               rd = new BufferedReader(new InputStreamReader(is, "UTF-8"));
 
                String factoryClassName = rd.readLine();
-               rd.close();
 
                if (factoryClassName != null && !"".equals(factoryClassName)) {
                    return newInstance(factoryClassName, classLoader);
                }
            }
        } catch (Exception ex) {
            LOGGER.log(Level.FINER, "Failed to load service " + factoryId + " from " + serviceId, ex);
-       }
-
-
+       } finally {
+           try {
+	       if (rd != null)
+		   rd.close();
+	   } catch (IOException ex) {
+	       LOGGER.log(Level.FINER, "Failed to close  BufferedReader.", ex);
+	   }
+	}
+
```

Comment:
If new InputStreamReader(is, "UTF-8") fails, then rd would be null but is would still be open. I would check is as well as rd in the finally clause.